### ISSUE TITLE
Potential issue with font size normalization on Google Pixel

### ISSUE THREAD

**Issue by Nickersoft** [react-native-training/react-native-elements#659]
### Description

**1. Explain what you did**
I ported over the font size normalization code for this library to my own app.

**2. What you expected to happen**
Correctly normalize my fonts on iOS and Android

**3. What actually happens**
iOS is fine, running on my device (2016 Google Pixel) results in impossibly small font sizes.

**Also please include the version of `react-native-elements` that you are using.**
I'm not.. just your code :)

### Solution

So I actually found the cause of the problem. The PixelRatio output on my device is 2.6, and the conditionals found in `normalizeText.js` only include whole numbers. As a result, the original fontSize was just being returned every time. I changed the code to this and it works:

```javascript
if (pixelRatio >= 2 && pixelRatio < 3) {
        // iphone 5s and older Androids
        if (deviceWidth < 360) {
            return size * 0.95;
        }
        // iphone 5
        if (deviceHeight < 667) {
            return size;
            // iphone 6-6s
        } else if (deviceHeight >= 667 && deviceHeight <= 735) {
            return size * 1.15;
        }
        // older phablets
        return size * 1.25;
    }
    else if (pixelRatio >= 3 && pixelRatio < 3.5) {
        // catch Android font scaling on small machines
        // where pixel ratio / font scale ratio => 3:3
        if (deviceWidth <= 360) {
            return size;
        }
        // Catch other weird android width sizings
        if (deviceHeight < 667) {
            return size * 1.15;
            // catch in-between size Androids and scale font up
            // a tad but not too much
        }
        if (deviceHeight >= 667 && deviceHeight <= 735) {
            return size * 1.2;
        }
        // catch larger devices
        // ie iphone 6s plus / 7 plus / mi note 等等
        return size * 1.27;
    }
    else if (pixelRatio >= 3.5) {
        // catch Android font scaling on small machines
        // where pixel ratio / font scale ratio => 3:3
        if (deviceWidth <= 360) {
            return size;
            // Catch other smaller android height sizings
        }
        if (deviceHeight < 667) {
            return size * 1.2;
            // catch in-between size Androids and scale font up
            // a tad but not too much
        }
        if (deviceHeight >= 667 && deviceHeight <= 735) {
            return size * 1.25;
        }
        // catch larger phablet devices
        return size * 1.4;
    }
    // if older device ie pixelRatio !== 2 || 3 || 3.5
    else return size;
```

I say this is a potential issue because the Snack example actually runs fine on my device, though I'm not sure why. Any insight as to why that may be the case would be appreciated, actually. I thought I'd bring this up though, seeing it does seem possible for pixel ratios to be numbers other than 2, 3, and 3.5.

**Comment by Monte9** [comment]
@Nickersoft your bring up a good point. And thanks for reporting this issue. We don't extensively test on each device size and so it's entirely possible that it's a bug for sure.

Your solution looks good to me. Would you mind submitting a PR for it?

**Comment by Nickersoft** [comment]
@Monte9 [How's this?](https://github.com/react-native-training/react-native-elements/pull/690)

**Comment by Monte9** [comment]
That's great! Thanks for the PR! It's been merged into `next` branch

It's part of `v0.18` which will be released today. Stay tuned

**Comment by iRoachie** [comment]
Fixed in v18.0.0
